protocols/virtualfund/virtualfund-single-hop_test.go

```diff
@@ -171,24 +149,58 @@ func TestSingleHopVirtualFund(t *testing.T) {
 			}
 
 			expectedSignedState := state.NewSignedState(o.V.PreFundState())
-			aliceSig, _ := o.V.PreFundState().Sign(my.privateKey)
-			_ = expectedSignedState.AddSignature(aliceSig)
+			mySig, _ := o.V.PreFundState().Sign(my.privateKey)
+			_ = expectedSignedState.AddSignature(mySig)
 
-			forBob := protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}}
-			forIrene := protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}}
-			want := protocols.SideEffects{MessagesToSend: []protocols.Message{forIrene, forBob}}
+			want := protocols.SideEffects{MessagesToSend: []protocols.Message{}}
+			switch my.role {
+			case 0:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+
+				}
+			case 1:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: alice.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+				}
+			case 2:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: alice.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+				}
+			}
+			// TODO ^^^^ the test is sensitive to the order of the messages. It should not be.
 
 			if diff := cmp.Diff(want, got); diff != "" {
 				if diff := cmp.Diff(want, got); diff != "" {
 					t.Errorf("TestCrank: side effects mismatch (-want +got):\n%s", diff)
 				}
 
 			}
+
 			// Manually progress the extended state by collecting prefund signatures
+			aliceSig, _ := vPreFund.Sign(alice.privateKey)
 			bobSig, _ := vPreFund.Sign(bob.privateKey)
 			p1Sig, _ := vPreFund.Sign(p1.privateKey)
-			o.V.AddStateWithSignature(vPreFund, bobSig)
-			o.V.AddStateWithSignature(vPreFund, p1Sig)
+			switch my.role {
+			case 0:
+				{
+					o.V.AddStateWithSignature(vPreFund, bobSig)
+					o.V.AddStateWithSignature(vPreFund, p1Sig)
+				}
+			case 1:
+				{
+					o.V.AddStateWithSignature(vPreFund, aliceSig) // TODO is this necessary?
```

Comment:
Is there a reason this wouldn't be necessary?

Reply:
If Alice had proposed V, then P1 and Bob will have the prefund signature *before* their objective is cranked for the first time -- going by our current engine implementation. That's why I was thinking it may not be necessary to add the prefund signature again: of course we are not driving the protocol with the engine in this test -- so the test order doesn't exactly match the order we expect. Thankfully though, we have designed the system to be very insensitive to the order of these things. So I will just remove these comments for now.